providers/cloudflare/cloudflareProvider.go

```diff
@@ -422,23 +423,46 @@ func newCloudflare(m map[string]string, metadata json.RawMessage) (providers.DNS
 
 // Used on the "existing" records.
 type cfRecData struct {
-	Name          string `json:"name"`
-	Target        string `json:"target"`
-	Service       string `json:"service"`       // SRV
-	Proto         string `json:"proto"`         // SRV
-	Priority      uint16 `json:"priority"`      // SRV
-	Weight        uint16 `json:"weight"`        // SRV
-	Port          uint16 `json:"port"`          // SRV
-	Tag           string `json:"tag"`           // CAA
-	Flags         uint8  `json:"flags"`         // CAA
-	Value         string `json:"value"`         // CAA
-	Usage         uint8  `json:"usage"`         // TLSA
-	Selector      uint8  `json:"selector"`      // TLSA
-	Matching_Type uint8  `json:"matching_type"` // TLSA
-	Certificate   string `json:"certificate"`   // TLSA
-	Algorithm     uint8  `json:"algorithm"`     // SSHFP
-	Hash_Type     uint8  `json:"type"`          // SSHFP
-	Fingerprint   string `json:"fingerprint"`   // SSHFP
+	Name          string      `json:"name"`
+	CFTarget      interface{} `json:"target"`
+	Service       string      `json:"service"`       // SRV
+	Proto         string      `json:"proto"`         // SRV
+	Priority      uint16      `json:"priority"`      // SRV
+	Weight        uint16      `json:"weight"`        // SRV
+	Port          uint16      `json:"port"`          // SRV
+	Tag           string      `json:"tag"`           // CAA
+	Flags         uint8       `json:"flags"`         // CAA
+	Value         string      `json:"value"`         // CAA
+	Usage         uint8       `json:"usage"`         // TLSA
+	Selector      uint8       `json:"selector"`      // TLSA
+	Matching_Type uint8       `json:"matching_type"` // TLSA
+	Certificate   string      `json:"certificate"`   // TLSA
+	Algorithm     uint8       `json:"algorithm"`     // SSHFP
+	Hash_Type     uint8       `json:"type"`          // SSHFP
+	Fingerprint   string      `json:"fingerprint"`   // SSHFP
+}
+
+func (c cfRecData) Target() string {
+	switch v := c.CFTarget.(type) {
+	case string:
+		return v
+	case bool:
+		if !v {
+			return "" // the "." is already added by nativeToRecord
```

Comment:
This is a slightly ugly hack, and will need to be kept in mind if ever using `Target()` in other places. To actually do this properly, nativeToRecord shouldn't add a `.` if it's already a FQDN. I'll fix this if it doesn't cause other side effects (I'm not too familiar with the cloudflare provider code yet).

Reply:
I've decided to leave this as-is, as the alternative would be to add the `.` directly in this function, which will either greatly increase the complexity of `Target()` and `GetTarget()` (checking for the trailing `.`, etc) or would cause them not to be the inverse of each other in all cases (which could introduce obscure bugs).